components/canbus.rst

```diff
@@ -0,0 +1,374 @@
+- **can_id** (*Optional*, int): Allows to override the can id configured in
+  the can bus device.
+- **use_extended_id** (*Optional*, boolean): default *False* identifies the type of *can_id*: 
+  *False*: Standard 11 Bit IDs, *True*: Extended 29Bit ID
+
+MCP2515
+-------
+
+The MCP2515 is a spi device and therfore you must first add the configuration for the spi bus to your file.
+You need to have an :ref:`SPI bus <spi>` in your configuration with both the **mosi_pin** and **miso_pin** set.
+
+For wireing up the MSP2515 please refer to the section below.
+
+- **cs_pin** (**Required**, :ref:`Pin Schema <config-pin_schema>`): Is used to tell the receiving SPI device
+  when it should listen for data on the SPI bus. Each device has an individual ``CS`` line.
+  Sometimes also called ``SS``.
+- **clock** (*Optional*): One of ``8MHZ``, ``16MHZ`` or ``20MHZ``. Clock crystal used on the MCP2515 device.
+  Defaults to ``8MHZ``.
+- **mode** (*Optional*): Operation mode. Default ot ``NORMAL``
+
+  - NORMAL: Normal operation
+  - LOOPBACK: Loopback mode can be used to just test you spi connections to the device
+  - LISTENONLY: only receive data
+
+.. code-block:: yaml
+
+    # Example configuration entry
+    canbus:
+      - platform: mcp2515
+        cs_pin: D5
+        can_id: 4
+        bit_rate: 50kbps
+        on_frame:
+        - can_id: 500
+            then:
+            - lambda: |-
+                std::string b(x.begin(), x.end());
+                ESP_LOGD("canid 500", "%s", &b[0] );
+            - light.turn_off: light_1
+        - can_id: 501
+            then:
+            - light.turn_on:
+                id: light_1
+                brightness: !lambda "return (float) x[0]/255;"
+        - can_id: 502
+            then:
+            - light.turn_on:
+                id: light_1
+                brightness: !lambda "return (float) x[0]/255;"
+
+Wireing options
+---------------
+Easiest approach is to just use fully assembled boards and just add one resistor in the MISO line.
+This runs MOSI, SCK and CS out of specification which is nearly never a problem.
+
+.. figure:: images/canbus_mcp2515_resistor.png
+    :align: center
+    :target: ../_images/canbus_mcp2515_resistor.png
+
+A more advanced option is to fully convert the 5V and 3.3V logic levels with a level shifter.
+
+.. figure:: images/canbus_mcp2515_txs0108e.png
+    :align: center
+    :target: ../_images/canbus_mcp2515_txs0108e.png
+
+
+Standard vs. Extended ID
+------------------------
+| Standard IDs and Extended IDs can coexist on the same segment.
+| It is important to know that for example Standard 0x123 and Extended 0x123 are different addesses.
+| This example shows how the different ID types are used in the configuration for transmission and receiving.
+| For the IDs decimal or hexadecimal notation is possible:
+| 0x000 - 0x7ff / 0-2047 for Standard IDs only.
+| 0x00000000 - 0x1fffffff / 0-536870911 for Extended IDs.
```

Comment:
Change format of this, you can use a note call out, please review the contributing docs for formatting

Reply:
Thanks for the hints, I will check them tomorrow, it's just time to go to bed in Germany ;-)

Reply:
OK, I kept the Chapter but renamed to "Extended ID" and made part of the Text a note.

"please review the contributing docs for formatting"
Are there other things to be optimized? I'm not exactly sure what you mean. Thanks.